connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/outbound/ConnectorJobHandler.java

```diff
@@ -231,6 +242,47 @@ public void handle(final JobClient client, final ActivatedJob job) {
     }
   }
 
+  private void handleFEELError(JobClient client, ActivatedJob job, ConnectorError error) {
+    if (error instanceof BpmnError bpmnError) {
+      LOGGER.debug("Throwing BPMN error for job {} with code {}", job.getKey(), bpmnError.code());
+      throwBpmnError(client, job, bpmnError);
+    } else if (error instanceof JobError jobError) {
+      LOGGER.debug("Throwing incident for job {}", job.getKey());
+      failJob(
+          client,
+          job,
+          new ErrorResult(
+              Map.of("error", jobError.message()),
+              new RuntimeException(jobError.message()),
+              jobError.retries(),
+              jobError.retryBackoff()));
+    }
+  }
+
+  private ConnectorResult handleSDKException(
+      ActivatedJob job,
+      Exception ex,
+      ConnectorRetryException.RetryPolicy retryPolicy,
+      String errorCode,
+      Duration retryBackoff) {
+    RetryContext retryContext =
+        tryGetRetryContext(job)
+            .map(ctx -> ctx.incrementAttemptedRetries(errorCode))
+            .orElse(new RetryContext(Map.of(errorCode, 0), job.getRetries(), retryBackoff));
+    RetryContext.RetryConfig retryConfig =
+        retryContext.computeNextRetryConfig(errorCode, retryPolicy);
+
+    return new ErrorResult(
+        Map.of(
+            "error",
+            exceptionToMap(ex),
+            ConnectorRetryException.RETRY_CONTEXT_INPUT_VARIABLE,
```

Comment:
We store the `retryContext` in the task's variables to retrieve the "state" on a later activation. 

> These variables become local process variables at the service task. They are passed along with that job when it is activated again, unless the worker fetches specific variables by name